apps/playground/app/test-data-list/page.tsx

```diff
@@ -0,0 +1,242 @@
+              </DataListRoot>
+            </Flex>
+            <Box mb="6">
+              <Heading mb="5" size="3">
+                With varied content
+              </Heading>
+              <DataListRoot direction={{ initial: 'column', sm: 'row' }} mt="9">
+                <DataListItem>
+                  <DataListLabel>Status!!!!!!!</DataListLabel>
+                  <DataListData>
+                    <Badge color="green" size="1" style={{ marginLeft: -2 }}>
+                      Active
+                    </Badge>
+                  </DataListData>
+                </DataListItem>
+
+                <DataListItem align="center">
+                  <DataListLabel>Name</DataListLabel>
+                  <DataListData>
+                    <Button size="1">Add</Button>
+                  </DataListData>
+                </DataListItem>
+
+                <DataListItem>
+                  <DataListLabel>Email</DataListLabel>
+                  <DataListData>vlad@workos.com</DataListData>
+                </DataListItem>
+
+                <DataListItem>
+                  <DataListLabel>Organization</DataListLabel>
+                  <DataListData>
+                    <Link href="https://workos.com">WorkOS</Link>
+                  </DataListData>
+                </DataListItem>
+
+                <DataListItem>
+                  <DataListLabel>Long value</DataListLabel>
+                  <DataListData>
+                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla ac nisl et libero
+                    ultricies viverra quis vitae quam. Proin a feugiat metus.
+                  </DataListData>
+                </DataListItem>
+
+                <DataListItem align="center">
+                  <DataListLabel>Authentication methods</DataListLabel>
+                  <DataListData>
+                    <Flex gap="2">
+                      <StarFilledIcon />
+                      <StarFilledIcon />
+                    </Flex>
+                  </DataListData>
+                </DataListItem>
+
+                <DataListItem>
+                  <DataListLabel>Long value</DataListLabel>
+                  <DataListData>
+                    Sed luctus, est id feugiat blandit, sapien nisl lobortis arcu, eu malesuada
+                    nulla ex ut lorem. In odio nisl, consectetur id commodo vel, posuere eu risus.
+                  </DataListData>
+                </DataListItem>
+              </DataListRoot>
+            </Box>
+
+            <Box mb="6">
+              <Heading mb="5" size="3">
+                With nested flex
+              </Heading>
+              <DataListRoot>
+                <DataListItem>
+                  <DataListLabel width={80}>Appearance</DataListLabel>
+                  <DataListData>
+                    <Flex align="center" gap="1">
+                      <IconButton size="1">
+                        <InfoCircledIcon />
+                      </IconButton>
```

Comment:
This should check for whether baseline alignment "just works" when data is a Flex with text that has a small icon or a small button aside.

Let's use `variant="ghost"` on the button the plain icon, otherwise a larger button pushes this element down and there is no expectation that this should work.